data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCrac.java

```diff
@@ -162,14 +189,16 @@ public void addState(State state) {
         Instant instant = getInstant(state.getInstant().getId());
 
         Optional<Contingency> stateContingency = state.getContingency();
+        Optional<Contingency> contingency;
         if (stateContingency.isPresent()) {
             if (contingencies.stream().noneMatch(stateContingency.get()::equals)) {
                 addContingency(stateContingency.get());
             }
```

Comment:
Do we want to check also the case when contingency are not equals but has the same ID, in order to throw an error if we are trying to create several contingencies with similar IDs ?

Reply:
It is already implemented in the addContingency or I didn't understand

Reply:
my bad